lwip-tests/patch/Common-allports.patch

```diff
@@ -0,0 +1,52 @@
+--- contrib_orginal/ports/Common.allports.mk	2024-01-09 16:16:39.398560526 +0100
++++ contrib/ports/Common.allports.mk	2024-01-10 16:38:01.764990843 +0100
+@@ -29,11 +29,10 @@
+ # Author: Adam Dunkels <adam@sics.se>
+ #
+ 
+-#CC=gcc
+-#CC=clang
++CC=$(CROSS)gcc
+ CCDEP?=$(CC)
+ 
+-CFLAGS+=-g -DLWIP_DEBUG -Wall -pedantic -Werror \
++CFLAGS+=-g -DLWIP_DEBUG -Wall -pedantic \
+ 	-Wparentheses -Wsequence-point -Wswitch-default \
+ 	-Wextra -Wundef -Wshadow -Wpointer-arith -Wcast-qual \
+ 	-Wc++-compat -Wwrite-strings -Wold-style-definition -Wcast-align \
+@@ -48,15 +47,15 @@
+ #CFLAGS:=$(CFLAGS) -fsanitize=address -fstack-protector -fstack-check -fsanitize=undefined -fno-sanitize=alignment
+ else
+ # we cannot sanitize alignment on x86-64 targets because clang wants 64 bit alignment
+-CFLAGS+= -fsanitize=address -fsanitize=undefined -fno-sanitize=alignment -Wdocumentation -Wno-documentation-deprecated-sync
++# CFLAGS+= -fsanitize=address -fsanitize=undefined -fno-sanitize=alignment -Wdocumentation -Wno-documentation-deprecated-sync
+ endif
+ 
+-CONTRIBDIR?=../../..
++CONTRIBDIR?=$(PREFIX_BUILD)/lwip-test/contrib-lwip
+ ARFLAGS?=rs
+ 
+ #Set this to where you have the lwip core module checked out from git
+ #default assumes it's a dir named lwip at the same level as the contrib module
+-LWIPDIR?=$(CONTRIBDIR)/../lwip/src
++LWIPDIR?=$(PREFIX_PROJECT)/phoenix-rtos-lwip/lib-lwip/src
+ 
+ CFLAGS+=-I. \
+ 	-I$(CONTRIBDIR) \
+@@ -64,7 +63,7 @@
+ 	-I$(LWIPARCH)/include
+ 
+ # Add include path and link to mbedTLS lib if available
+-MBEDTLSDIR?=$(CONTRIBDIR)/../mbedtls
++MBEDTLSDIR?=$(PREFIX_BUILD)/mbedtls/mbedtls-2.28.0
```

Comment:
It is not set

Reply:
to clarify: 
 - this line in patch is not needed ( `?=` in make means "set if not already set (eg. by ENV variable))
 - as we want to keep the patches minimal, please use ENV variable in build script instead of patching files whenever possible

Reply:
Got it